lib/blacklight/path.rb

```diff
@@ -0,0 +1,131 @@
+module Blacklight
+  # This class encapsulates the search state as represented by the query
+  # parameters namely: :f, :q, :page, :per_page and, :sort
+  class Path
```

Comment:
Should this go in `app/models/blacklight` now? Happy to be convinced either way.